src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -73,10 +143,10 @@ public TaskInfo createTask(Offer offer, TaskID taskId, NodeTask nodeTask) {
             nmTaskConfig.setAdvertisableCpus(profile.getCpus());
             nmTaskConfig.setAdvertisableMem(profile.getMemory());
             NodeManagerConfiguration nodeManagerConfiguration = this.cfg.getNodeManagerConfiguration();
-            nmTaskConfig.setUser(nodeManagerConfiguration.getUser().orNull());
             nmTaskConfig.setJvmOpts(nodeManagerConfiguration.getJvmOpts().orNull());
             nmTaskConfig.setCgroups(nodeManagerConfiguration.getCgroups().or(Boolean.FALSE));
             nmTaskConfig.setYarnEnvironment(cfg.getYarnEnvironment());
+            nmTaskConfig.setRemoteDistribution(cfg.getMyriadExecutorConfiguration().getNodeManagerUri().isPresent());
```

Comment:
Not sure I like this logic jump here. I'd rather the protocol/path specify whether it's remote or local, not the number/type of URIs present. Maybe the executor binary is remote, but the hadoop binary is local to each machine. Or vice versa. Or everything's local except the config. Let's break down the steps/actions and each step can decide how it responds to the uri type for its input(s).